Sources/XMTPiOS/Client.swift

```diff
@@ -197,8 +197,8 @@ public final class Client {
 	}
 
 	public static func createV3(account: SigningKey, options: ClientOptions) async throws -> Client {
-		let accountAddress = account.isSmartContractWallet ?
-			"eip155:\(String(describing: account.chainId)):\(account.address.lowercased())" :
+		let accountAddress = account.chainId != nil ?
```

Comment:
Do you not set the boolean for `isSmartContractWallet`? Do you only set the `chainId`? Wondering how we can make it more clear that the both need to be set if you're using a smart contract wallet. Though I guess we could get away with removing `isSmartContractWallet` potentially if you think the `chainId` will be enough? Or for ease of feature advancements we could change it to a `type` `EOA` or `SCW` etc...

Reply:
I think `chainId` is enough but I am not sure if dropping the `isSmartContract` leads to the clearest api. I can see people just seeing that the `chainId` field can be entered and passing it even with EOAs.

For RN I was thinking maybe an object might work but don't think that carries over to native so well. 

Could adopt the [chain-specific address](https://ethereum-magicians.org/t/chain-specific-addresses/6449) as it has become relatively common among SCA providers. So those including an `address` field of the form `{chainShortName}:{address}` get setup as SCA — although maybe this is less clear.

I wasn't actually judging the api though. I changed this is because the `chainId` needs to be unwrapped here & I just went with the condition from the `buildV3` method.